infrastructure-playbooks/purge-rgw-loadbalancer.yml

```diff
@@ -0,0 +1,113 @@
+---
+# This playbook purges Keepalived, loadbalancerProxy
+# It removes: packages, configuration files and ALL THE DATA
+#
+# Use it like this:
+# ansible-playbook purge-rgw-loadbalancer.yml
+#     Prompts for confirmation to purge, defaults to no and
+#     doesn't purge the cluster. yes purges the loadbalancer stuff.
+#
+# ansible-playbook -e ireallymeanit=yes|no purge-rgw-loadbalancer.yml
+#     Overrides the prompt using -e option. Can be used in
+#     automation scripts to avoid interactive prompt.
+
+- name: confirm whether user really meant to purge the loadbalancer stuff
+  hosts: localhost
+  gather_facts: false
+
+  vars_prompt:
+    - name: ireallymeanit
+      prompt: Are you sure you want to purge the loadbalancer stuff?
+      default: 'no'
+      private: no
+
+  tasks:
+  - name: exit playbook, if user did not mean to purge loadbalancer
+    fail:
+      msg: >
+        "Exiting purge-rgw-ha playbook, loadbalancer stuff was NOT purged.
+         To purge the loadbalancer stuff, either say 'yes' on the prompt or
+         or use `-e ireallymeanit=yes` on the command line when
+         invoking the playbook"
+    when: ireallymeanit != 'yes'
+
+- name: gather facts on all hosts
+
+  hosts:
+    - "{{ keepalived_group_name|default('keepaliveds') }}"
+    - "{{ haproxy_group_name|default('haproxys') }}"
+
+  become: true
+
+  tasks:
+    - debug: msg="gather facts on all loadbalancer hosts for following reference"
+
+- name: purge keepalived
+
+  vars:
+    keepalived_group_name: keepaliveds
+
+  hosts:
+    - "{{ keepalived_group_name|default('keepaliveds') }}"
+
+  gather_facts: false # Already gathered previously
+
+  become: true
+
+  tasks:
+
+  - name: stop keepalived with systemd
+    service:
+      name: keepalived
+      state: stopped
+      enabled: no
+    failed_when: false
+    when: ansible_service_mgr == 'systemd'
+
+  - name: remove config files
+    file:
+      path: "{{ item }}"
+      state: absent
+    with_items:
```

Comment:
You don't need the loop since you have a single item.

Reply:
oh, right, I shall fix all the "single" loops.